### PR TITLE
Refactor synapse features and move them to synapse from mediation

### PR DESCRIPTION
**Pull request by Vathsan**
Issue ID: https://github.com/wso2/micro-integrator/issues/638

### CODE REVIEW

File: modules/features/org.apache.synapse.samples.feature/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.synapse</groupId>
+            <artifactId>synapse-samples</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.synapse</groupId>
+            <artifactId>org.apache.synapse.wso2.feature</artifactId>
+            <version>2.1.7-wso2v96-SNAPSHOT</version>
```
Comment (Nirothipan) [inline modules/features/org.apache.synapse.samples.feature/pom.xml]: lets take this from project version

File: modules/features/pom.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>synapse-features</artifactId>
+    <packaging>pom</packaging>
+    <name>WSO2 Carbon - Synapse Feature Aggregator Module</name>
+    <url>http://wso2.org</url>
+
+    <modules>
+        <module>org.apache.synapse.wso2.feature</module>
+	    <module>org.apache.synapse.transport.fix.feature</module>
```
Comment (Nirothipan) [inline modules/features/pom.xml]: please do formatting

File: pom.xml [hunk 3]
```diff
@@ -516,11 +531,21 @@
...
+         </dependency>
          <dependency>
              <groupId>quickfixj</groupId>
              <artifactId>quickfixj-all</artifactId>
              <version>${qfj.version}</version>
          </dependency>
+         <dependency>
+             <groupId>quickfixj.wso2</groupId>
+             <artifactId>quickfixj</artifactId>
+             <version>${quickfixj.wso2.version}</version>
```
Comment (Nirothipan) [inline pom.xml]:
```suggestion
             <version>${qfj.version}.wso2</version>
```

File: modules/features/org.apache.synapse.samples.feature/pom.xml (hunk 1 above)
Comment (Vathsan) [reply modules/features/org.apache.synapse.samples.feature/pom.xml]: Done in https://github.com/wso2/wso2-synapse/pull/1391/commits/d9ab8a682e834be7903af1bfd95939f6014c9597

File: modules/features/pom.xml (hunk 2 above)
Comment (Vathsan) [reply modules/features/pom.xml]: Done in https://github.com/wso2/wso2-synapse/pull/1391/commits/d9ab8a682e834be7903af1bfd95939f6014c9597

**Pull request merged**